### PR TITLE
#135842533 Users to start a new game session

### PR DESCRIPTION
**Pull request by warenga**
#### What does this PR do?
Adds game.js model and api/games/:id/start endpoint.

 #### Description of Task to be completed?
To create an authenticated endpoint /api/games/:id/start to return the user as well as the friends that played in the searched game and the winner of that game.

 #### Any background context you want to provide?
I have also included an api/games endpoint that returns all games played.

 #### How should this be manually tested?
**_On postman_**
- Click `deployment` below.
- Copy the deployment url and paste it to postman

##### to search all games:
- Remove #!/ and replace it with api/games

##### to search a specific game (this requires a JWT token):
 _how to get a JWT token_
- Replace #!/ with api/auth/signin.
- Change to POST.
- In body, add email and password parameters and assign them to your authentication details.
- Click send. A token should be generated.

_how to use the token to access the endpoint:_
- Copy the token received (without the quotes).
- Replace api/auth/signin on the url with api/games/:id/start.
- Change to GET.
- Add the token copied as shown in the screenshot below.
- Set the id param to the gameID you are trying to search.

 #### What are the relevant pivotal tracker stories?
#135842533
 #### Screenshots (if appropriate)
screenshot of  /api/games/:id/start 
screen shot 2017-01-31 at 9 41 53 am

screenshot of api/games 
screen shot 2017-01-31 at 9 41 42 am

### CODE REVIEW

**Review by jackiewmacharia — APPROVED** [review]

File: app/controllers/game.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var config = require('../../config/config');
+var mongoose = require('mongoose');
+var Game = mongoose.model('Game');
+
+exports.all = function(req, res) {
+	Game.find({}).exec(function(err, game) {
+		if (err) return next(err);
+		res.send(game);
```
Comment (jkanyua) [inline app/controllers/game.js]: Think you should use res.json.
res.json eventually calls res.send, but before that it:

- respects the json spaces and json replacer app settings
- ensures the response will have utf8 charset and application/json content-type

File: config/routes.js [hunk 2]
```diff
@@ -72,6 +72,12 @@ module.exports = function(app, passport, auth) {
     //Finish with setting up the userId param
     app.param('userId', users.user);
 
+    // Game routes
+    var game = require('../app/controllers/game');
+    app.get('/api/games', game.all);
```
Comment (jkanyua) [inline config/routes.js:77]: Tested this on deployment using cURL& works
image

File: app/controllers/game.js (hunk 1 above)
Comment (warenga) [reply app/controllers/game.js]: Thanks for the observation. Changed it.

File: config/routes.js (hunk 2 above)
Comment (warenga) [reply config/routes.js:77]: great

File: config/routes.js [hunk 3]
```diff
@@ -72,6 +72,12 @@ module.exports = function(app, passport, auth) {
...
+    app.get('/api/games/:id/start', users.authToken, game.start);
```
Comment (jkanyua) [inline config/routes.js:78]: Wasn't able to test this on deployment. Please include screenshot of the GET (ID) on deployment.

**Review by jkanyua — APPROVED** [review]

File: config/routes.js (hunk 3 above)
Comment (gngchrs) [reply config/routes.js:78]: @andela-wmaina please test this and get back to me

Comment (warenga) [reply config/routes.js:78]: @andela-jkanyua please note that `/api/games/:id/start` is an authenticated endpoint that requires a JWT token. Probably why it was not working. Thank you for pointing this out though, I have added a much more descriptive step-by-step explanation of how to obtain the token and how to use it. I have also replaced the screenshot with the deployed version.

Comment (warenga) [reply config/routes.js:78]: @gangachris I have tested it and added the screenshots in the PR comment.

**Pull request merged**